packages/phone-number-privacy/src/index.ts

```diff
@@ -1,96 +1,12 @@
+import { handleGetContactMatches } from './match-making/get-contact-matches'
+import { handleGetSalt } from './salt-generation/get-salt'
 
 export const getSalt = functions.https.onRequest(async (request, response) => {
-  // TODO (amyslawson) refactor this internal logic and input validation to its own file
-  // when adding error handling
-  try {
-    const queryQuota: QueryQuota = new QueryQuota()
-    if (!isValidGetSaltInput(request.body)) {
-      response.status(400).send('Invalid input parameters')
-      return
-    }
-    authenticateUser()
-    const remainingQueryCount = await queryQuota.getRemainingQueryCount(
-      request.body.account,
-      request.body.phoneNumber
-    )
-    if (remainingQueryCount <= 0) {
-      response.status(400).send('Requester exceeded salt service query quota')
-      return
-    }
-    const salt = computeBLSSalt(request.body.queryPhoneNumber)
-    await incrementQueryCount(request.body.account).catch((error) => {
-      // TODO [amyslawson] think of failure case here
-      console.error(error)
-    })
-    response.json({ success: true, salt })
-  } catch (e) {
-    console.log('Failed to getSalt', e)
-    response.status(500).send(e)
-  }
+  await handleGetSalt(request, response)
 })
 
-function isValidGetSaltInput(requestBody: any): boolean {
-  return (
-    hasValidAccountParam(requestBody) &&
-    hasValidPhoneNumberParam(requestBody) &&
-    hasValidQueryPhoneNumberParam(requestBody)
-  )
-}
-
-function hasValidAccountParam(requestBody: any): boolean {
-  return requestBody.account && (requestBody.account as string).startsWith('0x')
-}
-
-function hasValidPhoneNumberParam(requestBody: any): boolean {
-  return requestBody.phoneNumber && PhoneNumberUtils.isE164Number(requestBody.phoneNumber)
-}
-
-function hasValidQueryPhoneNumberParam(requestBody: any): boolean {
-  return requestBody.queryPhoneNumber
-}
-
 // TODO (amyslawson) consider pagination or streaming of contacts?
 export const getContactMatches = functions.https.onRequest(async (request, response) => {
-  // TODO (amyslawson) refactor this internal logic and input validation to its own file
-  // when adding error handling
-  try {
-    if (!isValidGetContactMatchesInput(request.body)) {
-      response.status(400).send('Invalid input parameters')
-      return
-    }
-    authenticateUser()
-    const matchedContacts: ContactMatch[] = (
-      await getNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
-    ).map((numberPair) => ({ phoneNumber: numberPair, salt: computeBLSSalt(numberPair) }))
-    await setNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
-    // TODO (amyslawson) return salts with contact
-    response.json({ success: true, matchedContacts })
-  } catch (e) {
-    console.log('Failed to getContactMatches', e)
-    response.status(500).send(e)
-  }
+  await handleGetContactMatches(request, response)
```

Comment:
No need to await if it's the only thing in this method. Or to put it another way, there's no benefit to awaiting if something has no lines after it.